backend/windmill-api/src/database_triggers/trigger.rs

```diff
@@ -0,0 +1,474 @@
+use std::{collections::HashMap, pin::Pin};
+
+use crate::{
+    database_triggers::{
+        get_database_resource,
+        relation::RelationConverter,
+        replication_message::{
+            LogicalReplicationMessage::{Begin, Commit, Delete, Insert, Relation, Type, Update},
+            ReplicationMessage,
+        },
+        run_job,
+    },
+    db::DB,
+    users::fetch_api_authed,
+};
+use bytes::{BufMut, Bytes, BytesMut};
+use chrono::TimeZone;
+use futures::{pin_mut, SinkExt, StreamExt};
+use pg_escape::{quote_identifier, quote_literal};
+use rand::seq::SliceRandom;
+use rust_postgres::{Client, Config, CopyBothDuplex, NoTls, SimpleQueryMessage};
+use windmill_common::{db::UserDB, worker::to_raw_value, INSTANCE_NAME};
+
+use super::{
+    handler::{Database, DatabaseTrigger},
+    replication_message::PrimaryKeepAliveBody,
+};
+
+pub struct LogicalReplicationSettings {
+    pub streaming: bool,
+}
+
+impl LogicalReplicationSettings {
+    pub fn new(streaming: bool) -> Self {
+        Self { streaming }
+    }
+}
+
+#[allow(unused)]
+trait RowExist {
+    fn row_exist(&self) -> bool;
+}
+
+impl RowExist for Vec<SimpleQueryMessage> {
```

Comment:
nit: if used only once it might be prefered to just use a function instead `fn row_exist(vec: &Vec<SimpleQueryMessage>) -> bool`